test/functional/legacy/autocmd_option_spec.lua

```diff
@@ -0,0 +1,128 @@
+local helpers = require('test.functional.helpers')
+local clear, nvim, eq = helpers.clear, helpers.nvim, helpers.eq
+local source, execute = helpers.source, helpers.execute
+
+describe('au OptionSet', function()
+  setup(clear)
+
+  describe('with * as <amatch>', function()
+    describe('matches when being set any option', function()
+
+      local function expected_str(option, oldval, newval, scope)
+        return ''
+          .. string.format('Autocmd Option: <%s>,', option)
+          .. string.format(' OldVal: <%s>,', oldval)
+          .. string.format(' NewVal: <%s>,', newval)
+          .. string.format(' Scope: <%s>', scope)
+      end
+
+      local function get_result()
+        return nvim('get_var', 'ret')
+      end
+
+      local function expected_combination(option, oldval, newval, scope)
+        eq(expected_str(option, oldval, newval, scope), get_result())
+      end
+
+      local function expected_empty()
+        eq('', get_result())
+      end
+
+      setup(function()
+
+        source([[
+          fu! AutoCommand(match)
+            let g:ret.=printf('Autocmd Option: <%s>,', a:match)
+            let g:ret.=printf(' OldVal: <%s>,', v:option_old)
+            let g:ret.=printf(' NewVal: <%s>,', v:option_new)
+            let g:ret.=printf(' Scope: <%s>', v:option_type)
+          endfu
+          
+          au OptionSet * :call AutoCommand(expand("<amatch>"))
+        ]])
+      end)
+
+      before_each(function()
+        execute([[let g:ret = '']])
```

Comment:
Should not it also contain `reset` call? This way I believe you are using same nvim instance for all tests which is not good because busted actually expects tests to be isolated (and it supports an option to shuffle them to check that tests are isolated).